packages/snowpack/src/build/build-pipeline.ts

```diff
@@ -110,43 +121,53 @@ async function runPipelineLoadStep(
 async function runPipelineTransformStep(
   output: SnowpackBuildMap,
   srcPath: string,
-  {plugins, messageBus, isDev, sourceMaps}: BuildFileOptions,
+  {plugins, isDev, logLevel = 'info', sourceMaps}: BuildFileOptions,
 ): Promise<SnowpackBuildMap> {
   const srcExt = getExt(srcPath).baseExt;
   const rootFileName = path.basename(srcPath).replace(srcExt, '');
   for (const step of plugins) {
     if (!step.transform) {
       continue;
     }
-    for (const destExt of Object.keys(output)) {
-      const destBuildFile = output[destExt];
-      const {code} = typeof destBuildFile === 'string' ? {code: destBuildFile} : destBuildFile;
-      const result = await step.transform({
-        contents: code,
-        fileExt: destExt,
-        filePath: rootFileName + destExt,
-        isDev,
-        // @ts-ignore: internal API only
-        log: (msg, data: any = {}) => {
-          messageBus.emit(msg, {
-            ...data,
-            id: step.name,
-            msg: data.msg && `[${srcPath}] ${data.msg}`,
-          });
-        },
-        // @ts-ignore: Deprecated
-        urlPath: `./${path.basename(rootFileName + destExt)}`,
-      });
 
-      // if step returned a value, only update code (don’t touch .map)
-      if (typeof result === 'string') {
-        output[destExt].code = result;
-      } else if (result && typeof result === 'object' && (result as {result: string}).result) {
-        output[destExt].code = (result as {result: string}).result;
-      }
+    const pluginLogger =
+      pluginLoggers[step.name] || createLogger({name: step.name, level: logLevel});
+    pluginLoggers[step.name] = pluginLogger;
+
+    try {
+      for (const destExt of Object.keys(output)) {
+        const destBuildFile = output[destExt];
+        const {code} = typeof destBuildFile === 'string' ? {code: destBuildFile} : destBuildFile;
+        const filePath = rootFileName + destExt;
+        const debugPath = path.relative(process.cwd(), filePath);
+        pluginLogger.debug(`transform() starting… [${debugPath}]`);
+        const result = await step.transform({
+          contents: code,
+          fileExt: destExt,
+          filePath,
+          isDev,
+          // @ts-ignore: internal API only
+          log: (msg, data: any = {}) => {
```

Comment:
now that we have a good logger, we can pass it directly to the different plugin hooks as our documented, public way to log. Since these are internal only at the moment, no worries about breaking changes.

Reply:
I like that. Also using a documented logger like Pino seems easy-to-use, too.